fcrepo-kernel-modeshape/src/main/java/org/fcrepo/kernel/modeshape/FedoraBinaryImpl.java

```diff
@@ -100,18 +100,38 @@ public FedoraBinaryImpl(final Node node) {
 
     @Override
     public FedoraResource getDescription() {
+        final Node descNode = getDescriptionNode();
+        if (descNode == null) {
+            return null;
+        }
         return new NonRdfSourceDescriptionImpl(getDescriptionNode());
     }
 
     @Override
     protected Node getDescriptionNode() {
         try {
+            final Node node = getNode();
+            if (isMemento()) {
+                final String mementoName = node.getName();
+                return node.getNode("../../" + FEDORA_DESCRIPTION + "/" + LDPCV_TIME_MAP + "/" + mementoName);
```

Comment:
This makes me cringe, also isn't the Binary just one `../` in front of the `FEDORA_DESCRIPTION`?

Reply:
Yeah its pretty cringy. I started writing it with a pile of `getParent`s and `getNode`s, but it didn't feel more readable to me.  I'm fine with changing it if others prefer though.

Reply:
At a minimum, a detailed comment explaining the structure that requires `../../` would help in future maintenance.

Reply:
I see you have taken a different approach here:
https://github.com/fcrepo4/fcrepo4/pull/1344/files#diff-7aee50d5030524028026aa94cd75f350R419

Reply:
It was a shorter path :) If its preferable I can switch this to 
```
return node.getParent().getParent().getNode(FEDORA_DESCRIPTION)
                        .getNode(LDPCV_TIME_MAP).getNode(mementoName);
```
And over in `NonRdfSourceDescriptionImpl.getContentNode` to
```
return node.getParent().getParent().getParent().getNode(LDPCV_TIME_MAP).getNode(mementoName);
```
(feel sorry for anyone that ends up unit testing that)

Reply:
Actually, decided to save us another round of review buttons and switched https://github.com/fcrepo4/fcrepo4/pull/1344/commits/2fe2a0e9048afe6df4ae756abb1e220995bce4f7